### PR TITLE
Fix back button handling

### PR DESCRIPTION
**Pull request by adasiewiczr**

### CODE REVIEW

**Comment by adasiewiczr** [comment]
CI is failing because version `4.3.1-beta-3` is not available yet. Tested with composite build

File: samplekotlin/src/main/java/com/example/chattutorial/ChannelActivity.kt [hunk 1]
```diff
@@ -44,12 +44,21 @@ class ChannelActivity : AppCompatActivity(R.layout.activity_channel) {
...
+
+        // step 4 - handle navigate up state
+        messageListViewModel.state.observe(this) {
+            when (it) {
+                is MessageListViewModel.State.NavigateUp -> finish()
+                else -> Unit
+            }
+        }
+
+        // step 5 - let the message input know when we are editing a message
```
Comment (tschellenbach) [inline samplekotlin/src/main/java/com/example/chattutorial/ChannelActivity.kt:56]: is there no way to simplify this back button handling stuff? @JcMinarro thoughts?
it takes up too much of this part of the tutorial. we're trying to explain how to build chat, not how to handle a back button, it's a bit distracting

**Review by tschellenbach — APPROVED** [review]
its a fix, good to merge it, we need a better solution though to simplify this activity...

File: samplekotlin/src/main/java/com/example/chattutorial/ChannelActivity.kt (hunk 1 above)
Comment (carterhudson) [reply samplekotlin/src/main/java/com/example/chattutorial/ChannelActivity.kt:56]: Is there a reason mode isn't part of state?

**Pull request merged**